File: wagerproof-ios-native/Wagerproof/Features/Analytics/Components/LRSplitsSection.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// L/R Pitcher Splits: notable matchups (indigo accent) first, then all
/// other splits. Mirrors RN `LRSplitsBody`/`SplitRow`.
struct LRSplitsSection: View {
    let splits: [MLBLRSplitEntry]

    var body: some View {
        let notable = splits.filter { $0.isNotable }
        let rest = splits.filter { !$0.isNotable }

        VStack(alignment: .leading, spacing: 16) {
            if !notable.isEmpty {
                VStack(alignment: .leading, spacing: 8) {
                    RegressionGroupLabel(
                        label: "NOTABLE MATCHUPS",
                        count: notable.count,
                        color: Regression.accentIndigo,
                        note: "Favorable or difficult splits worth flagging"
                    )
                    ForEach(Array(notable.enumerated()), id: \.offset) { _, split in
                        LRSplitRow(split: split, notable: true)
                    }
                }
            }
            if !rest.isEmpty {
                VStack(alignment: .leading, spacing: 6) {
                    RegressionGroupLabel(label: "ALL OTHER SPLITS", count: rest.count)
                    ForEach(Array(rest.enumerated()), id: \.offset) { _, split in
                        LRSplitRow(split: split, notable: false)
                    }
                }
            }
        }
    }
}

private struct LRSplitRow: View {
    let split: MLBLRSplitEntry
    let notable: Bool

    var body: some View {
        let record = split.f5Ties > 0
            ? "F5 \(split.f5Wins)-\(split.f5Losses)-\(split.f5Ties)"
            : "F5 \(split.f5Wins)-\(split.f5Losses)"
        let winColor: Color = {
            guard let pct = split.f5WinPct else { return Color.appTextPrimary }
            if pct >= 60 { return Regression.winGreen }
            if pct <= 40 { return Regression.lossRed }
            return Color.appTextPrimary
        }()

        RegressionAccentRow(color: notable ? Regression.accentIndigo : .clear) {
            HStack(alignment: .top, spacing: 10) {
                VStack(alignment: .leading, spacing: 1) {
                    HStack(spacing: 4) {
                        Text(split.teamName)
                            .font(.system(size: 13, weight: .bold))
                            .foregroundStyle(Color.appTextPrimary)
                        Text(split.facing)
                            .font(.system(size: 13, weight: .medium))
                            .foregroundStyle(Color.appTextSecondary)
                    }
                    Text("vs \(split.opponentSp ?? split.opponent) (\(split.opponentSpHand)HP)")
                        .font(.system(size: 11))
                        .foregroundStyle(Color.appTextSecondary)
                        .lineLimit(1)
                }
                Spacer(minLength: 0)
                VStack(alignment: .trailing, spacing: 2) {
                    Text("\(Regression.trimmed(split.avgF5Runs)) R/G")
                        .font(.system(size: 13, weight: .semibold))
                        .monospacedDigit()
                        .foregroundStyle(Color.appTextPrimary)
                    Text(record + (split.f5WinPct.map { "  \(Regression.trimmed($0))%" } ?? ""))
                        .font(.system(size: 11, weight: .semibold))
                        .monospacedDigit()
                        .foregroundStyle(winColor)
                }
            }
        }
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/analytics/LRSplitsSection.kt
package com.wagerproof.app.features.analytics

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBLRSplitEntry

/**
 * L/R Pitcher Splits: notable matchups (indigo accent) first, then all other
 * splits. Port of iOS `LRSplitsSection` / RN `LRSplitsBody`.
 */
@Composable
fun LRSplitsSection(splits: List<MLBLRSplitEntry>, modifier: Modifier = Modifier) {
    val notable = splits.filter { it.isNotable }
    val rest = splits.filter { !it.isNotable }

    Column(modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(16.dp)) {
        if (notable.isNotEmpty()) {
            Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                RegressionGroupLabel(
                    label = "NOTABLE MATCHUPS",
                    count = notable.size,
                    color = Regression.accentIndigo,
                    note = "Favorable or difficult splits worth flagging",
                )
                notable.forEach { LRSplitRow(it, notable = true) }
            }
        }
        if (rest.isNotEmpty()) {
            Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                RegressionGroupLabel(label = "ALL OTHER SPLITS", count = rest.size)
                rest.forEach { LRSplitRow(it, notable = false) }
            }
        }
    }
}

@Composable
private fun LRSplitRow(split: MLBLRSplitEntry, notable: Boolean) {
    val record = if (split.f5Ties > 0) {
        "F5 ${split.f5Wins}-${split.f5Losses}-${split.f5Ties}"
    } else {
        "F5 ${split.f5Wins}-${split.f5Losses}"
    }
    val winColor: Color = split.f5WinPct?.let { pct ->
        when {
            pct >= 60 -> Regression.winGreen
            pct <= 40 -> Regression.lossRed
            else -> AppColors.appTextPrimary
        }
    } ?: AppColors.appTextPrimary

    RegressionAccentRow(color = if (notable) Regression.accentIndigo else Color.Transparent) {
        Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.Top, horizontalArrangement = Arrangement.spacedBy(10.dp)) {
            Column(Modifier.weight(1f)) {
                Row {
                    Text(split.teamName, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = AppColors.appTextPrimary)
                    Text("  ${split.facing}", fontSize = 13.sp, fontWeight = FontWeight.Medium, color = AppColors.appTextSecondary)
                }
                Text(
                    text = "vs ${split.opponentSp ?: split.opponent} (${split.opponentSpHand}HP)",
                    fontSize = 11.sp,
                    color = AppColors.appTextSecondary,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
            Column(horizontalAlignment = Alignment.End) {
                Text(
                    text = "${Regression.trimmed(split.avgF5Runs)} R/G",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = FontFamily.Monospace,
                    color = AppColors.appTextPrimary,
                )
                Text(
                    text = record + (split.f5WinPct?.let { "  ${Regression.trimmed(it)}%" } ?: ""),
                    fontSize = 11.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = FontFamily.Monospace,
                    color = winColor,
                )
            }
        }
    }
}
